### ISSUE TITLE
Truncate check HTTP check output if too long

### ISSUE THREAD

**Issue by jakubgs** [hashicorp/consul#8611]
### Feature Description

I'd like the output from HTTP checks to be truncated if it's too long. Either automatically or via configuration option that specifies number of lines/characters for example.

#### Use Case(s)

Many services don't provide a sensible health check enfpoint, which often results in having to use `/` or `/index.html` as healt check, but that can often result in a massive HTML response that makes the Consul UI for checks harder to navigate due to the size.

**Comment by jakubgs** [comment]
My main point is about very large output, like HTML or similar. Your proposed solution is good, and would cover most cases, but just the first line is sometimes not optimal.

For example, lets say my healthcheck for a [Netadata](https://www.netdata.cloud/) instance is the `/api/v1/alarms` endpoint:
```
 % curl -s localhost:8001/api/v1/alarms
{
	"hostname": "some-host.example.org",
	"latest_alarm_log_unique_id": 1633014848,
	"status": false,
	"now": 1633983766,
	"alarms": {

	}
}
```
As we can see the output is a short formatted JSON that tells me whether any alarms are triggered. This is 9 lines long, and provides useful information in addition to being a healthcheck. In that case I wouldn't mind seeing some of the output, since just one line would only contain the first `{` character.

It's a tricky balance, one line can be fine for many services, but several lines would be convenient for quite a few. But then again the need to click "See more..." is a mild inconvenience compared to having to scroll through hundreds or thousands of lines of HTML to get to the next healhcheck in the list so I'd take it.

**Comment by jkirschner-hashicorp** [comment]
I agree that choosing an `N` (in the range of 10-16) and calling it a day is the right approach for now. I think this would be a good candidate for a community contribution once we have a mock-up.

If we find that some users have more health checks than we expected, we could consider changing the layout to fit more checks on screen at once - thanks for including those quick mocks here!

Out of curiosity, what's the maximum number of checks you've had at once? And how many of them had multi-line output?

**Comment by jkirschner-hashicorp** [comment]
Hi @samanvithbalugari,

Just checking in to see how things are going and whether you have any questions!

**Comment by jkirschner-hashicorp** [comment]
Hi @samanvithbalugari,

Checking in again. Let us know how we can support your contribution (e.g., answer your questions).

And if your circumstances have changed and you're no longer able to take this on at the moment, that's totally fine, just let us know so I can open it up for others to work on. Thanks!

**Comment by jkirschner-hashicorp** [comment]
Hi @samanvithbalugari,

Since I haven't heard from you in a while, I'm unassigning you in Github so that other community members can take this task up if interested. Let us know if you're interested in working on this again. Thanks!
